packages/functions/src/oidc/awsCredentialsProvider.ts

```diff
@@ -0,0 +1,59 @@
+import type { AwsCredentialIdentityProvider } from '@smithy/types';
+import type { FromWebTokenInit } from '@aws-sdk/credential-provider-web-identity';
+import { getVercelOidcToken } from './getVercelOidcToken';
+
+/**
+ * The init object for the `awsCredentialsProvider` function.
+ */
+export type AwsCredentialsProviderInit = Omit<
+  FromWebTokenInit,
+  'webIdentityToken'
+>;
+
+/**
+ * Obtains the Vercel OIDC token and creates an AWS credential provider function
+ * that gets AWS credentials calling STS AssumeRoleWithWebIdentity API.
+ *
+ * ```javascript
+ * import * as s3 from '@aws-sdk/client-s3';
+ * import { awsCredentialsProvider } from '@vercel/functions/oidc';
+ *
+ * const s3Client = new s3.S3Client({
```

Comment:
why not use `@example` and the rest of jsdocs stuff 🙂

Reply:
I made the docs on this as familiar as possible, it matches the same style as the AWS SDK jsdoc comments

Reply:
it could be both, in this way https://github.com/vercel/vercel/pull/11701/files#diff-36e729cab01b983624e1459052aaf4d59277379ca0e4567322f6a808ba4cd66cR6